### DISCUSSION TITLE
How to identify whether the form rendered due to reset or not?

### DISCUSSION THREAD

**Discussion by nanivijay** [react-hook-form/react-hook-form#4754] [Q&A]
I am trying to use reset api to set the default values after the async api call. How to identify whether the values are set using reset or not? SO that i can add a condition to the useEffect to execute some specific logic during the reset.

Any ideas?

**Comment by bluebill1049** [comment]
I think you will manage that state at your own app, to track it's set via `reset` API or defaultValues
